src/sentry/tsdb/snuba.py

```diff
@@ -193,6 +211,32 @@ class SnubaTSDB(BaseTSDB):
     def __init__(self, **options):
         super().__init__(**options)
 
+    def __manual_group_on_time_aggregation(self, rollup, time_column_alias) -> Sequence[Any]:
```

Comment:
This basically mirrors what snuba has here https://github.com/getsentry/snuba/blob/a8c0b5c10807ef981ed3e8deb58e924a575f6814/snuba/query/processors/logical/timeseries_processor.py#L110

Error and transaction tables has a `TimeSeriesProcessor` which added syntactic sugar on a reserved `time` column that boils down to re-writing the query when the query hit snuba. Going foward, we don't want to rely on `TimeSeriesProcessor` being available on `search_issues` table, so we must apply an aggregation here in-place of what snuba would have done after processing our raw query.

Reply:
lgtm. I wonder if we need our own abstraction that sits in between us and snuba that allows us to add these "virtual" columns or something. I know that @ceorourke needs this in a separate area as well, and will need to duplicate this code.

Reply:
Yeah, that's a good idea. @ceorourke could you point out where in your work you'll be needing this. I'll start thinking about where to put this abstraction.

Reply:
Oh I didn't mean specifically for this query - colleen will be using snql instead of this old format. I was more wondering if we need a replacement for things like `TimeSeriesProcessor` that exist on our end rather than on the snuba side.

Reply:
As far as I'm aware, `TimeSeriesProcessor` is the only thing we need to work around. I'm sure there will be other stuff I missed though. So this abstraction is similar to an extra processing step for SnQL queries to 'fill in the gap' on common stuff we'll need?

Reply:
Are you referring to stuff like this: https://github.com/getsentry/sentry/blob/828c5fcdd8d7c1f9d1241d5f06c768ffd5cbcafc/src/sentry/issues/query.py#L9-L11

Reply:
Yeah, just ended up being fairly painful to add things like that everywhere.